bundles/org.openhab.binding.meteofrance/src/main/java/org/openhab/binding/meteofrance/internal/handler/RainForecastHandler.java

```diff
@@ -0,0 +1,155 @@
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.meteofrance.internal.handler;
+
+import static org.openhab.binding.meteofrance.internal.MeteoFranceBindingConstants.*;
+import static org.openhab.core.types.TimeSeries.Policy.REPLACE;
+
+import java.time.ZonedDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.meteofrance.internal.config.ForecastConfiguration;
+import org.openhab.binding.meteofrance.internal.dto.RainForecast;
+import org.openhab.binding.meteofrance.internal.dto.RainForecast.Forecast;
+import org.openhab.binding.meteofrance.internal.dto.RainForecast.Properties;
+import org.openhab.binding.meteofrance.internal.dto.RainForecast.RainIntensity;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.PointType;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.openhab.core.types.TimeSeries;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link RainForecastHandler} is responsible for updating channels
+ * and querying the API
+ *
+ * @author Gaël L'hopital - Initial contribution
+ */
+@NonNullByDefault
+public class RainForecastHandler extends BaseThingHandler implements MeteoFranceChildHandler {
+    private final Logger logger = LoggerFactory.getLogger(RainForecastHandler.class);
+    private final ChannelUID intensityChannelUID;
+
+    private Optional<ScheduledFuture<?>> refreshJob = Optional.empty();
+    private Optional<PointType> location = Optional.empty();
+
+    public RainForecastHandler(Thing thing) {
+        super(thing);
+        this.intensityChannelUID = new ChannelUID(getThing().getUID(), INTENSITY);
+    }
+
+    @Override
+    public @Nullable Bridge getBridge() {
+        return super.getBridge();
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Rain Forecast handler.");
+        disposeJob();
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        ForecastConfiguration config = getConfigAs(ForecastConfiguration.class);
```

Comment:
No need to check the configuration ? For like `isBlank()` or for having a numeric value? (i assumed the department code are all numeric ?)
and set the thing state to `CONFIGURATION_ERROR` accordingly?